### ISSUE TITLE
Resolve parser defects

### ISSUE THREAD

**Issue by automaticp** [nolanderc/glsl_analyzer#15]
I'll be dumping snippets that confuse/crash the parser here.

Here's a first couple (v1.1.0):

```glsl
#version 330 core

out Interface {
    vec4 frag_pos;
} out_block;

void main() {
    out_block.frag_pos = gl_in[0].gl_Position; // <- expected ;
}
```

```glsl
#version 330 core
void main() {
    if (true) { discard; } // <- expected }
}
```

**Comment by automaticp** [comment]
The first one seems to be confusing `.f` as a floating point number.

The second one is unclear, but the parser does not recover well after it and spits out errors in the rest of the file if there's more code. All is fine without the braces though:
```glsl
if (true) discard; // fine
```

**Comment by nolanderc** [comment]
Thank you for taking the time to report these!

The above snippets should be solved in #19 and #20 and released in v1.1.2.

I'm going to close this issue, if you find any more cases, just open a new issue! :-)
